spock-specs/src/test/groovy/org/spockframework/smoke/mock/JavaStubs.groovy

```diff
@@ -160,20 +160,84 @@ class JavaStubs extends Specification {
     person.phoneNumber == "6789"
   }
 
+  @Issue("https://github.com/spockframework/spock/issues/2039")
   def "cannot stub final methods with byteBuddy"() {
+    given:
+    FinalMethodPerson person = Stub(mockMaker: MockMakers.byteBuddy)
+
+    when:
+    person.getPhoneNumber() >> 6789
+
+    then:
+    def ex = thrown(InvalidSpecException)
+    ex.message == "The final method 'getPhoneNumber' can't be mocked by the byte-buddy mock maker. Please use another mock maker supporting final methods."
+
+    expect:
+    person.getPhoneNumber() == "12345"
+  }
+
+  @Issue("https://github.com/spockframework/spock/issues/2039")
+  def "cannot stub final methods with byteBuddy without error message when one overload is non final"() {
```

Comment:
It fails to mock the method,. because one of the overloads is final, so there is not error message, because the parameters could have matched both the final or the non-final method, so we can't detect that right now => No error message.

This is the best-effort part of this PR.

Reply:
Yes, but this doesn't become obvious. The confusing part is what `without error message` refers to. It can be read in two ways: one, that you can't do this without getting an error message, or it is a scenario where you don't expect to get an error message.